pennylane/ops/functions/equal.py

```diff
@@ -99,3 +114,14 @@ def equal(
                 return False
 
     return getattr(op1, "inverse", False) == getattr(op2, "inverse", False)
+
+
+def equal_measurements(op1, op2):
+    """Determine whether two MeasurementProcess objects are equal"""
+    return_types_match = op1.return_type == op2.return_type
+    observables_match = op1.obs == op2.obs
+    wires_match = op1.wires == op2.wires
+
+    if return_types_match and observables_match and wires_match:
```

Comment:
Do I need to only check wires_match if wires is not None? I have it this way because I believe if `wires` is None for both then they would be equal and `wires_match` would be True.

Reply:
If `wires` is not `None` then `obs` has to be `None` and the other way around.

However we always need to check for `return_types_match`, because we can have:
```python
qml.equal(qml.counts(wires=0), qml.probs(wires=0))
```